### PR TITLE
feat(compute): add compute regional template create 

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/CreateRegionalInstanceTemplate.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.concurrent.TimeoutException;
+
+public class CreateRegionalInstanceTemplate {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the instance you want to delete.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/CreateRegionalInstanceTemplate.java:41]: create?

File: compute/cloud-client/src/test/java/compute/SnippetsIT.java [hunk 2]
```diff
@@ -105,6 +109,10 @@ public static void setUp()
     compute.CreateInstance.createInstance(PROJECT_ID, ZONE, MACHINE_NAME_WAIT_FOR_OP);
     compute.CreateEncryptedInstance
         .createEncryptedInstance(PROJECT_ID, ZONE, MACHINE_NAME_ENCRYPTED, RAW_KEY);
+    CreateRegionalInstanceTemplate
+        .createRegionalInstanceTemplate(PROJECT_ID, REGION, REGIONAL_LOCATION_NAME);
+    assertThat(stdOut.toString()).contains("Instance Template Operation Status: DONE");
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/SnippetsIT.java:114]: Kindly refrain from using stdOut to assert. Instead make an API call to assert if the instance is created. Applies to multiple places.

File: compute/cloud-client/src/test/java/compute/SnippetsIT.java [hunk 3]
```diff
@@ -130,6 +138,11 @@ public static void cleanup()
     compute.DeleteInstance.deleteInstance(PROJECT_ID, ZONE, MACHINE_NAME);
     compute.DeleteInstance.deleteInstance(PROJECT_ID, ZONE, MACHINE_NAME_LIST_INSTANCE);
     compute.DeleteInstance.deleteInstance(PROJECT_ID, ZONE, MACHINE_NAME_WITH_SSD);
+    DeleteRegionalInstanceTemplate
+        .deleteRegionalInstanceTemplate(PROJECT_ID, REGION, REGIONAL_LOCATION_NAME);
+    assertThat(stdOut.toString())
+        .contains("Instance template deletion operation status for "
+            + REGIONAL_LOCATION_NAME);
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/SnippetsIT.java:146]: ditto

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/CreateRegionalInstanceTemplate.java (hunk 1 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/CreateRegionalInstanceTemplate.java]: Thank you! Fixed.

File: compute/cloud-client/src/test/java/compute/SnippetsIT.java (hunk 2 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/SnippetsIT.java]: Thank you! Creation is tested in testGetRegionalInstanceTemplate, deletion is tested in ReservationIT class.

**Review by Sita04 — APPROVED** [review]
LGTM. Thanks

**Pull request merged**